clients/grpc-client/src/main/java/org/projectnessie/client/grpc/GrpcClientBuilder.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.client.grpc;
+
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import org.projectnessie.client.NessieClient;
+
+/**
+ * A builder class that creates a {@link NessieGrpcClient} via {@link GrpcClientBuilder#builder()}.
+ */
+public class GrpcClientBuilder {
+
+  private String endpoint;
+
+  /**
+   * Returns a new {@link GrpcClientBuilder} instance.
+   *
+   * @return A new {@link GrpcClientBuilder} instance.
+   */
+  public static GrpcClientBuilder builder() {
+    return new GrpcClientBuilder();
+  }
+
+  /**
+   * Sets the Nessie gRPC Server endpoint.
+   *
+   * @param endpoint The gRPC Server endpoint
+   * @return {@code this}
+   */
+  public GrpcClientBuilder withEndpoint(String endpoint) {
+    this.endpoint = endpoint;
```

Comment:
Any reason why this cannot be a URI so HttpClientBuilder + GrpcClientBuilder can share that property (and have a commit AbstractClientBuilder class)?

Reply:
made `endpoint` an `URI` but I don't see any good reason for introducing an `AbstractClientBuilder`